Casks/z/zen-browser@twilight.rb

```diff
@@ -0,0 +1,37 @@
+cask "zen-browser@twilight" do
+  arch arm: "aarch64", intel: "x86_64"
+
+  version "1.0.1-t.20"
+  sha256 :no_check
+
+  url "https://github.com/zen-browser/desktop/releases/download/twilight/zen.macos-#{arch}.dmg",
+      verified: "github.com/zen-browser/desktop/"
+  name "Zen Twilight"
+  desc "Gecko based web browser"
+  homepage "https://zen-browser.app/"
+
+  livecheck do
+    url "https://api.github.com/repos/zen-browser/desktop/releases/tags/twilight"
+    regex(/Twilight\s+build\s+-\s+v?(\d+(?:\.\d+)+(?:[.-][a-z]\.\d+)?)/i)
+    strategy :json do |json, regex|
+      match = json["name"]&.match(regex)
+      next if match.blank?
+
+      match[1]
+    end
```

Comment:
@khipp The general approach makes sense to me but we avoid using GitHub API URLs like this, as we don't yet have a way of making those requests using `HOMEBREW_GITHUB_API_TOKEN` like `GithubLatest` and `GithubReleases` do. The unauthenticated rate limit is only 60 requests per hour, so that could become a problem if we allowed checks like this.

I should be able to make some changes to the `GithubReleases` strategy to allow it to handle a `releases/tags/...` API URL and that would resolve the issue for this particular use case. To allow this to move forward in the interim time, we can have this check the releases page (searching for "twilight"):

```suggestion
    url "https://github.com/zen-browser/desktop/releases?q=twilight"
    regex(%r{/tag/twilight[^>]+?>.+?v?(\d+(?:\.\d+)+(?:[._-][a-z]\.\d+)?)}i)
    strategy :page_match
```

This isn't ideal but it avoids any rate-limit issues for the time being and we can rework this to use a modified version of your approach if/when the `GithubReleases` changes are available.

Reply:
That makes sense to me, thank you!

Reply:
I forgot to mention it but if you're interested in a bit of livecheck history: we used to have GitHub API URLs in `livecheck` blocks in the olden days (before 2020, when we moved them from Homebrew/homebrew-livecheck into homebrew/core formulae). Our CI setup in the homebrew-livecheck repository was rudimentary, so we would often run into rate limit issues on bulk PRs that modified a lot of `livecheck` blocks (or if I was on a roll and creating a lot of PRs in a short period of time). Due to this, we stopped using GitHub API URLs and switched to checking GitHub releases pages (which we used until mid-2023 when we created the `Github*` strategies). That's just to say that this restriction comes from past suffering that we don't want to repeat 😆

Side note: the `#livecheckable?` method gets its name from the old "livecheckable" terminology. Originally livecheck information was stored in separate files in the homebrew-livecheck repository (e.g., from [my fork](https://github.com/samford/homebrew-livecheck/commit/241770a63d31a50794c3d7e0294d0fe31861c762)) and those files were referred to as "livecheckables". At this point, the `#livecheckable?` method tells us whether there's a `livecheck` block but the method name is sometimes confusing to anyone who doesn't know this ancient history (i.e., they may think the method tells us whether livecheck can check the package/resource). I should probably get around to changing the method name to `#livecheck_block?` with a `#livecheckable?` alias for compatibility.

Reply:
Thank you! It's always great to see how projects evolve over time and the long history of trial and error behind them. It seems like you all have really picked up the pace over the last few years.

Reply:
> It seems like you all have really picked up the pace over the last few years.

That's for sure. I started working on livecheck in late 2019 but there wasn't much activity at the time (some maintainers had moved on in 2018). The pace accelerated in early 2020 when I became a maintainer (and immediately focused on improving livecheck), followed by the Google Summer of Code project to integrate livecheck into brew and homebrew-core (Nanda was our student, I was the primary mentor, and we had a _very_ busy summer), and we merged cask support near the end of the year. A lot of the foundation for what livecheck is today was laid in 2020.

Adoption increased with better integration, support, and visibility (PRs in our main repositories instead of homebrew-livecheck), and eventually it became an essential tool for maintainers/contributors. That was the goal from the start but it took thousands of PRs to get to that point (not to mention ongoing maintenance, as you well know 😆). There's still plenty that I want/need to accomplish (my to-do list is never-ending) but I think it's fair to say that livecheck's in better shape than it was, thanks to everyone who has worked on it over the years (now including you).